src/controller/CommissionerDiscoveryController.cpp

```diff
@@ -455,6 +460,10 @@ void CommissionerDiscoveryController::InternalHandleContentAppPasscodeResponse()
                     cd, Transport::PeerAddress::UDP(client->GetPeerAddress().GetIPAddress(), client->GetCdPort()));
                 return;
             }
+            // Store commissioner passcode as setup PIN (string/charspan).
+            // If this PIN is not empty it signals that user prompt was shown to enter PIN and AccountLogin::Login command has to be sent.
+            mCommissionerSetupPin = CharSpan::fromCharString(std::to_string(passcode));
```

Comment:
I don't understand how this compiles either.  That's causing the `std::string` here to be converted to a `const char *` and how?

But again, this looks like a use-after-return.

Reply:
Yes, I think the better approach here is to just store the uint passcode, and convert on demand when necessary, to avoid user-after-return with the spans.

Reply:
"avoiding use after return" is a mandatory/critical thing, indeed...

Reply:
Code removed, fixed.